plugins/main/public/components/management/cluster/dashboard/dashboard.tsx

```diff
@@ -0,0 +1,206 @@
+  nodesCount: number;
+  agentsCount: number;
+}
+
+const DashboardCT: React.FC<DashboardCTProps> = ({ statusRunning }) => {
+  const {
+    filters,
+    dataSource,
+    fetchFilters,
+    isLoading: isDataSourceLoading,
+    fetchData,
+    setFilters,
+  } = useDataSource<tParsedIndexPattern, PatternDataSource>({
+    DataSource: AlertsDataSource,
+    repository: new AlertsDataSourceRepository(),
+  });
+
+  const [state, setState] = useState<ClusterDashboardState>({
+    showConfig: false,
+    showNodes: false,
+    nodeList: [],
+    configuration: undefined,
+    version: undefined,
+    nodesCount: 0,
+    agentsCount: 0,
+  });
+
+  const [results, setResults] = useState<SearchResponse>({} as SearchResponse);
+
+  const { searchBarProps } = useSearchBar({
+    indexPattern: dataSource?.indexPattern as IndexPattern,
+    filters,
+    setFilters,
+  });
+  const { query, dateRangeFrom, dateRangeTo } = searchBarProps;
+
+  useEffect(() => {
+    if (isDataSourceLoading) {
+      return;
+    }
+    fetchData({
+      query,
+      dateRange: {
+        from: dateRangeFrom,
+        to: dateRangeTo,
+      },
+    })
+      .then(results => {
+        setResults(results);
+      })
+      .catch(error => {
+        const searchError = ErrorFactory.create(HttpError, {
+          error,
+          message: 'Error fetching vulnerabilities',
+        });
+        ErrorHandler.handleError(searchError);
+      });
+  }, [JSON.stringify(fetchFilters), JSON.stringify(query)]);
+
+  const setBooleans = (component: string | null) => {
+    setState({
+      ...state,
+      showConfig: component === 'showConfig',
+      showNodes: component === 'showNodes',
+    });
+  };
+
+  const goAgents = () => {
+    getCore().application.navigateToApp(endpointSummary.id, {
+      path: '#/agents-preview',
+    });
+  };
+
+  const goBack = () => {
+    setBooleans(null);
+  };
+
+  const goNodes = () => {
+    setBooleans('showNodes');
+  };
+
+  const goConfiguration = () => {
+    setBooleans('showConfig');
+  };
+
+  useEffect(() => {
+    const getData = async () => {
+      const data = await Promise.all([
+        WzRequest.apiReq('GET', '/cluster/nodes', {}),
+        WzRequest.apiReq('GET', '/cluster/local/config', {}),
+        WzRequest.apiReq('GET', '/', {}),
+        WzRequest.apiReq('GET', '/agents', { limit: 1 }),
+        WzRequest.apiReq('GET', '/cluster/healthcheck', {}),
+      ]);
+
+      const nodeList = ((data[0] || {}).data || {}).data || {} || false;
+      const clusterConfig = ((data[1] || {}).data || {}).data || {} || false;
+      const agents = ((data[3] || {}).data || {}).data || {} || false;
+      setState({
+        ...state,
+        configuration: clusterConfig.affected_items[0],
+        version: (((data[2] || {}).data || {}).data || {}).api_version || false,
+        nodesCount: nodeList.total_affected_items,
+        agentsCount: agents.total_affected_items - 1,
+        nodeList: nodeList?.affected_items ?? [],
```

Comment:
**suggestion**: this could be enhanced using the optional chaining operator `?.`.